addons/medical_engine/functions/fnc_getItemArmor.sqf

```diff
@@ -18,53 +18,42 @@
+        private _itemInfo = configFile >> "CfgWeapons" >> _item >> "ItemInfo";
+        private _itemType = getNumber (_itemInfo >> "type");
 
-    if (_itemType == TYPE_UNIFORM) then {
-        private _unitCfg = configFile >> "CfgVehicles" >> getText (_itemInfo >> "uniformClass");
-        if (_hitpoint == "#structural") then {
-            // TODO: I'm not sure if this should be multiplied by the base armor value or not
-            _armor = getNumber (_unitCfg >> "armorStructural");
+        if (_itemType == TYPE_UNIFORM) then {
+            private _unitCfg = configFile >> "CfgVehicles" >> getText (_itemInfo >> "uniformClass");
+            if (_hitpoint == "#structural") then {
+                // TODO: I'm not sure if this should be multiplied by the base armor value or not
+                _armor = getNumber (_unitCfg >> "armorStructural");
+            } else {
+                private _entry = _unitCfg >> "HitPoints" >> _hitpoint;
+                _armor = getNumber (_unitCfg >> "armor") * (1 max getNumber (_entry >> "armor"));
+                _passThrough = 0.1 max getNumber (_entry >> "passThrough") min 1; // prevent dividing by 0
+            };
         } else {
-            private _entry = _unitCfg >> "HitPoints" >> _hitpoint;
-            _armor = getNumber (_unitCfg >> "armor") * (1 max getNumber (_entry >> "armor"));
-            _passThrough = 0.1 max getNumber (_entry >> "passThrough") min 1; // prevent dividing by 0
+            private _condition = format ["getText (_x >> 'hitpointName') == '%1'", _hitpoint];
+            private _entry = configProperties [_itemInfo >> "HitpointsProtectionInfo", _condition] param [0, configNull];
+            if (!isNull _entry) then {
+                _armor = getNumber (_entry >> "armor");
+                _passThrough = 0.1 max getNumber (_entry >> "passThrough") min 1;
+            };
         };
-    } else {
-        private _condition = format ["getText (_x >> 'hitpointName') == '%1'", _hitpoint];
-        private _entry = configProperties [_itemInfo >> "HitpointsProtectionInfo", _condition] param [0, configNull];
-        if (!isNull _entry) then {
-            _armor = getNumber (_entry >> "armor");
-            _passThrough = 0.1 max getNumber (_entry >> "passThrough") min 1;
-        };
-    };
 
-    // Scale armor using passthrough to fix explosive-resistant armor (#9063)
-    // Skip scaling for uniforms and items that don't cover the hitpoint to prevent infinite armor
-    if (_armor > 0) then {
-        if (_itemType == TYPE_UNIFORM) then {
-            _armorScaled = _armor;
-        } else {
-            _armorScaled = (log (_armor / (_passThrough ^ _passThroughEffect))) * 10;
+        // Scale armor using passthrough to fix explosive-resistant & stupid armor (#9063)
+        // Skip scaling for uniforms and items that don't cover the hitpoint to prevent infinite armor
+        private _armorLevelStep = [4, 2] select (_itemType == TYPE_HEADGEAR);
```

Comment:
Doesn't this change the balance compared to before?

I haven't looked at it in detail, so forgive me if I'm wrong.